### PR TITLE
New macro: Change scene to yards

### PR DESCRIPTION
**Pull request by AncAinu**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Translation (updates existing localization strings or adds complete new language file)
- [ ] Release (administrative update for tagged release, eg, changelog, readme, manifest)
- [ ] Maintenance (updates to the repository, dependencies and other non-functional code management)

## Checklist:

- [ ] I have run linting on my code to follow the follow the [style](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/blob/dev/.eslintrc.json) of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new unhandled or unexpected warnings.
- [x] My change requires an update to the [documentation](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/wiki).
- [ ] I can be reached on Discord by my handle: ___

## Related Issue(s) 

Fixes or addresses #issue(s):

## Description

Change scene grid to 2 yards, keeping lighting at scale.

### Motivation and context

Useful for maps created for DnD that have a grid of 5ft, or any other grid size.

### Development / Testing Environment
- Foundry VTT: v9

### CODE REVIEW

**Comment by AncAinu** [comment]
I did it for Foundry V9, might not work on v10 (but I'm sure requires just little change).
I'm happy for you to do, or tell me what to do to make this a viable PR.

**Comment by Jagusti** [comment]
Thanks for the suggested macro ... I really appreciate you following the macro format I'm using. 

I can see this maybe fitting into a toolbox of canvas macros ... something for me to consider in context of a few others 🤔 Certainly timely for the upcoming Light and Vision release. 

As it stands, this won't work on v10 because of the [data model changes](https://foundryvtt.com/article/v10-data-model/). There are minor amends needed to strip out `data` in the object path and switch `gridDistance` to `grid.distance`.  If you get a chance to do it before me, great. If not, can I ask you to just make sure I can make edits to the contribution. Thanks.

File: scripts/macros/change-scene-to-yards.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const scale = 2 / canvas.scene.data.gridDistance;
+await canvas.scene.update({
+"gridDistance": 2,
+"gridUnits": "yd"
+});
+const newLights = canvas.scene.data.lights.map(light => {
```
Comment (Jagusti) [inline scripts/macros/change-scene-to-yards.js]: Should now be `canvas.scene.lights`

File: scripts/macros/change-scene-to-yards.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  return {
+    _id: light.id,
+    "config.dim": light.data.config.dim * scale,
+    "config.bright": light.data.config.bright * scale
```
Comment (Jagusti) [inline scripts/macros/change-scene-to-yards.js]: Also now `light.config`

File: scripts/macros/change-scene-to-yards.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const scale = 2 / canvas.scene.data.gridDistance;
```
Comment (Jagusti) [inline scripts/macros/change-scene-to-yards.js]: Should be `canvas.scene.grid.distance`

**Review by Jagusti — CHANGES_REQUESTED** [review]
See https://foundryvtt.com/article/v10-data-model/ for v10 data model changes that this should use

File: scripts/macros/change-scene-to-yards.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+await canvas.scene.update({
+"gridDistance": 2,
```
Comment (Jagusti) [inline scripts/macros/change-scene-to-yards.js]: Should be `grid.distance`

File: scripts/macros/change-scene-to-yards.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+"gridUnits": "yd"
```
Comment (Jagusti) [inline scripts/macros/change-scene-to-yards.js]: Should be `grid.units`

**Comment by AncAinu** [comment]
@Jagusti I made the update for V10, also I did use the script in V10, so I can confirm it works well.

**Review by Jagusti — APPROVED** [review]

**Pull request merged**